packages/core/src/components/slider/handle.tsx

```diff
@@ -48,7 +48,10 @@ export class Handle extends AbstractPureComponent<IHandleProps, IHandleState> {
 
     private handleElement: HTMLElement;
     private refHandlers = {
-        handle: (el: HTMLSpanElement) => (this.handleElement = el),
+        handle: (el: HTMLSpanElement) => {
+            this.handleElement = el;
+            this.forceUpdate();
```

Comment:
This is here because when the number of handles changes, these handles are re-instantiated. When that happens the first render of the component does not have the ref captured and thus incorrectly positions the handle. This fixes by immediately re-rendering once the ref is captured, not sure if there's a nicer way to handle (haha) this. Otherwise, will add a comment to this effect.